model.py

```diff
@@ -4836,24 +4836,55 @@ def scaled_weight(self):
-        Subject.AXIS_360_AUDIENCE : 30,
-        (DataSource.OVERDRIVE, Subject.GRADE_LEVEL) : 20,
-        (DataSource.OVERDRIVE, Subject.INTEREST_LEVEL) : 20,
-        Subject.OVERDRIVE : 15,
-        (DataSource.AMAZON, Subject.AGE_RANGE) : 10,
-        (DataSource.AMAZON, Subject.GRADE_LEVEL) : 9,
-
-        # These measure reading level, except for TAG, which measures
-        # who-knows-what.
-        (DataSource.OVERDRIVE, Subject.GRADE_LEVEL) : 5,
-        Subject.TAG : 2,
-        Subject.LEXILE_SCORE : 1,
-        Subject.ATOS_SCORE: 1,
-
-        Subject.AGE_RANGE : 10,
-        Subject.GRADE_LEVEL : 10,
+        
+        # Not all classifications are equally reliable as indicators
+        # of a target age. This dictionary contains the coefficients
+        # we multiply against the weights of incoming classifications
+        # to reflect the overall reliability of that type of
+        # classification.
+        #
+        # If we had a ton of information about target age this might
+        # not be necessary--it doesn't seem necessary for genre
+        # classifications. But we sometimes have very little
+        # information about target age, so being careful about how
+        # much we trust different data sources can become important.
+        
+        DataSource.MANUAL : 1.0,
+        DataSource.LIBRARY_STAFF: 1.0,        
+        (DataSource.METADATA_WRANGLER, Subject.AGE_RANGE) : 1.0,
+
+        Subject.AXIS_360_AUDIENCE : 0.9,
+        (DataSource.OVERDRIVE, Subject.INTEREST_LEVEL) : 0.9,
+        (DataSource.OVERDRIVE, Subject.OVERDRIVE) : 0.9, # But see below
+        (DataSource.AMAZON, Subject.AGE_RANGE) : 0.9,
+        (DataSource.AMAZON, Subject.GRADE_LEVEL) : 0.9,
+        
+        # Although Overdrive usually reserves Fiction and Nonfiction
+        # for books for adults, it's not as reliable an indicator as
+        # other Overdrive classifications.
+        (DataSource.OVERDRIVE, Subject.OVERDRIVE, "Fiction") : 0.7,
+        (DataSource.OVERDRIVE, Subject.OVERDRIVE, "Nonfiction") : 0.7,
+        
+        Subject.AGE_RANGE : 0.6,
+        Subject.GRADE_LEVEL : 0.6,
+        
+        # There's no real way to know what this measures, since it
+        # could be anything. If a tag mentions a target age or a grade
+        # level, the accuracy seems to be... not terrible.
+        Subject.TAG : 0.45,
+
+        # Tags that come from OCLC Linked Data are of lower quality
+        # because they sometimes talk about completely the wrong book.
+        (DataSource.OCLC_LINKED_DATA, Subject.TAG) : 0.3,
+        
+        # These measure reading level, not age appropriateness.
+        # However, if the book is a remedial work for adults we won't
+        # be calculating a target age in the first place, so it's okay
+        # to use reading level as a proxy for age appropriateness in a
+        # pinch. (But not outside of a pinch.)
+        (DataSource.OVERDRIVE, Subject.GRADE_LEVEL) : 0.35,
```

Comment:
I'm undoing here a change I made recently, which I made because I forgot that Overdrive's "grade level" classifications are measurements of reading level.